kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/EngineServiceDiscovery.scala

```diff
@@ -33,6 +34,10 @@ class EngineServiceDiscovery(
   override def stop(): Unit = synchronized {
     if (!isServerLost.get()) {
       discoveryClient.deregisterService()
+      val path = s"/metrics${conf.get(HA_NAMESPACE)}"
```

Comment:
The code appears to concat this path multiple times. I suggest using a method to call.